airbyte-workers/src/main/java/io/airbyte/workers/helper/FailureHelper.java

```diff
@@ -95,6 +95,18 @@ public static AttemptFailureSummary failureSummaryForCancellation(final Long job
     return failureSummary(failures, partialSuccess);
   }
 
+  public static AttemptFailureSummary failureSummaryForTemporalCleaningJobState(final Long jobId, final Integer attemptNumber) {
+    final FailureReason failureReason = new FailureReason()
+        .withFailureOrigin(FailureOrigin.AIRBYTE_PLATFORM)
+        .withFailureType(FailureType.SYSTEM_ERROR)
+        .withInternalMessage(
+            "Setting attempt to FAILED because the temporal workflow for this connection was restarted, and existing job state was cleaned.")
+        .withExternalMessage("Airbyte internal connection workflow was restarted, causing incomplete jobs to fail. No user action required.")
```

Comment:
@andyjih do you have any feedback on the copy here? For context, this situation happens if the temporal workflow that underlies a connection dies while a job is running; when the workflow is restarted, it will mark any non-terminal jobs as failed before creating any new jobs - this marking of orphaned jobs as failed is where this external message will be used. Here is an example screenshot showing what this looks like to the user:
<img width="1363" alt="Screen Shot 2022-05-05 at 4 20 58 PM" src="https://user-images.githubusercontent.com/22731524/167042597-1d9015a5-2230-4eea-9aaa-a6ce50906392.png">

Reply:
Gotcha! What you have totally works, but I wrote a couple options below just in case.

1. "An internal Airbyte error has occurred. Airbyte will attempt to run this job again."
2. "Airbyte connection workflow has restarted. No user action required."

Reply:
Thank you for your input! I like the first option you provided - I don't really expect users to understand what a "connection workflow" is so its probably best to leave that out, and I like that it states that we will run the job again.